File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int n=nums.size();
        int low=0;
        int high=n-1;
        int res=n;             // n isliye lerhe hain cuz agar nhi aya kuch to whi ans aajaega 

        while(low<=high){
            int guess=(low+high)/2;

            if(nums[guess]>=target){
                res=guess;    // ye wala ho skta hai target ka index
                high=guess-1;      // Left mein jao
            }
            else{
                low=guess+1;       // Right mei jao 
            }
        }
        return res;            // Index final wla return kardo 
    }
};

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        n=len(nums)
        low,high,res=0,n-1,n

        while low<=high:
            guess=(low+high)//2

            if nums[guess]>=target:
                res=guess
                high=guess-1
            else:
                low=guess+1
        return res